--- /mnt/portage/portage/x11-drivers/nvidia-drivers/nvidia-drivers-304.37-r1.ebuild	2012-08-21 22:01:31.000000000 -0700
+++ nvidia-drivers-304.37-r1.ebuild	2012-08-22 21:13:33.716791429 -0700
@@ -22,7 +22,7 @@
 LICENSE="NVIDIA"
 SLOT="0"
 KEYWORDS="-* ~amd64 ~x86 ~amd64-fbsd ~x86-fbsd"
-IUSE="acpi multilib kernel_FreeBSD kernel_linux +tools +X"
+IUSE="acpi multilib kernel_FreeBSD kernel_linux pax_kernel +tools +X"
 RESTRICT="strip"
 EMULTILIB_PKG="true"
 
@@ -139,6 +139,12 @@
 		# If greater than 2.6.5 use M= instead of SUBDIR=
 		convert_to_m "${NV_SRC}"/Makefile.kbuild
 	fi
+
+	if use pax_kernel; then
+	    epatch "${FILESDIR}"/nvidia-drivers-pax-const.patch
+	    epatch "${FILESDIR}"/nvidia-drivers-pax-usercopy.patch
+	fi
+
 	cat <<- EOF > "${S}"/nvidia.icd
 		/usr/$(get_libdir)/libnvidia-opencl.so
 	EOF
