--- /mnt/portage/portage/x11-drivers/nvidia-drivers/nvidia-drivers-295.49.ebuild	2012-05-04 07:58:57.000000000 -0700
+++ nvidia-drivers-295.49.ebuild	2012-05-05 11:52:00.676444319 -0700
@@ -19,7 +19,7 @@
 LICENSE="NVIDIA"
 SLOT="0"
 KEYWORDS="-* ~amd64 ~x86 ~x86-fbsd"
-IUSE="acpi custom-cflags multilib kernel_linux +tools"
+IUSE="acpi custom-cflags multilib kernel_linux +tools pax_kernel"
 RESTRICT="strip"
 EMULTILIB_PKG="true"
 
@@ -299,6 +299,12 @@
 		# If greater than 2.6.5 use M= instead of SUBDIR=
 		convert_to_m "${NV_SRC}"/Makefile.kbuild
 	fi
+
+	if use pax_kernel; then
+	    epatch "${FILESDIR}"/nvidia-drivers-pax-const.patch
+	    epatch "${FILESDIR}"/nvidia-drivers-pax-usercopy.patch
+	fi
+
 	cat <<- EOF > "${S}"/nvidia.icd
 		/usr/$(get_libdir)/libcuda.so
 	EOF
