--- ../../../../portage/x11-drivers/nvidia-drivers/nvidia-drivers-290.06.ebuild	2011-11-04 11:11:42.000000000 -0700
+++ nvidia-drivers-290.10.ebuild	2011-11-24 03:28:04.027569344 -0800
@@ -19,7 +19,7 @@
 LICENSE="NVIDIA"
 SLOT="0"
 KEYWORDS="-* ~amd64 ~x86 ~x86-fbsd"
-IUSE="acpi custom-cflags gtk multilib kernel_linux"
+IUSE="acpi custom-cflags gtk multilib kernel_linux pax_kernel"
 RESTRICT="strip"
 EMULTILIB_PKG="true"
 
@@ -288,6 +288,11 @@
 		# If greater than 2.6.5 use M= instead of SUBDIR=
 		convert_to_m "${NV_SRC}"/Makefile.kbuild
 	fi
+
+	if use pax_kernel; then
+	    epatch "${FILESDIR}"/nvidia-drivers-pax-const.patch
+	    epatch "${FILESDIR}"/nvidia-drivers-pax-usercopy.patch
+	fi
 }
 
 src_compile() {
