--- /mnt/portage/portage/x11-drivers/nvidia-drivers/nvidia-drivers-295.53.ebuild	2012-05-21 16:31:31.000000000 -0700
+++ nvidia-drivers-295.53.ebuild	2012-05-22 08:59:13.433939065 -0700
@@ -21,7 +21,7 @@
 LICENSE="NVIDIA"
 SLOT="0"
 KEYWORDS="-* ~amd64 ~x86 ~amd64-fbsd ~x86-fbsd"
-IUSE="acpi multilib kernel_FreeBSD kernel_linux +tools"
+IUSE="acpi multilib kernel_FreeBSD kernel_linux pax_kernel +tools"
 RESTRICT="strip"
 EMULTILIB_PKG="true"
 
@@ -303,6 +303,12 @@
 		# If greater than 2.6.5 use M= instead of SUBDIR=
 		convert_to_m "${NV_SRC}"/Makefile.kbuild
 	fi
+
+	if use pax_kernel; then
+	    epatch "${FILESDIR}"/nvidia-drivers-pax-const.patch
+	    epatch "${FILESDIR}"/nvidia-drivers-pax-usercopy.patch
+	fi
+
 	cat <<- EOF > "${S}"/nvidia.icd
 		/usr/$(get_libdir)/libcuda.so
 	EOF
