--- /mnt/portage/portage/app-emulation/wine/wine-1.5.10.ebuild	2012-08-07 01:06:43.000000000 -0700
+++ wine-1.5.10.ebuild	2012-08-07 09:31:02.394262211 -0700
@@ -133,7 +133,8 @@
 	epatch "${FILESDIR}"/${PN}-1.1.15-winegcc.patch #260726
 	epatch "${FILESDIR}"/${PN}-1.4_rc2-multilib-portage.patch #395615
 	epatch "${FILESDIR}"/${PN}-1.5.10-osmesa-check.patch #429386
-	epatch "${DISTDIR}/${PULSE_PATCH}" #421365
+	# epatch "${DISTDIR}/${PULSE_PATCH}" #421365
+	epatch "${FILESDIR}"/pulseaudio.patch
 	epatch_user #282735
 	eautoreconf
 	sed -i '/^UPDATE_DESKTOP_DATABASE/s:=.*:=true:' tools/Makefile.in || die
@@ -178,7 +179,7 @@
 		$(use_with oss) \
 		$(use_with png) \
 		$(use_with threads pthread) \
-		$(use_with pulseaudio pulse) \
+		$(use_with pulseaudio) \
 		$(use_with scanner sane) \
 		$(use_enable test tests) \
 		$(use_with truetype freetype) \
