--- theHarvester-ng-blackhat.orig/discovery/dnssearch-threads.py	2011-07-27 08:45:21.000000000 -0700
+++ theHarvester-ng-blackhat/discovery/dnssearch-threads.py	2011-12-07 20:57:26.916821679 -0800
@@ -63,7 +63,7 @@
 	def __init__(self,domain,dnsserver,verbose=False):
 		self.domain=domain
 		self.server = dnsserver
-		self.file="dns-names.txt"
+		self.file="/usr/share/theHarvester/dns-names.txt"
 		self.subdo = False 
 		self.verbose = verbose
 		try:
diff -ur theHarvester-ng-blackhat.orig/discovery/dnssearch.py theHarvester-ng-blackhat/discovery/dnssearch.py
--- theHarvester-ng-blackhat.orig/discovery/dnssearch.py	2011-07-27 08:45:21.000000000 -0700
+++ theHarvester-ng-blackhat/discovery/dnssearch.py	2011-12-07 20:57:07.196822097 -0800
@@ -61,7 +61,7 @@
 	def __init__(self,domain,dnsserver,verbose=False):
 		self.domain=domain
 		self.nameserver = dnsserver
-		self.file="dns-names.txt"
+		self.file="/usr/share/theHarvester/dns-names.txt"
 		self.subdo = False 
 		self.verbose = verbose
 		try:
